docs/new_protocol.asciidoc

```diff
@@ -0,0 +1,463 @@
+	tcptuple := testTcpTuple()
+	req := protos.Packet{Payload: data1}
+	resp := protos.Packet{Payload: data2}
+
+	private := protos.ProtocolData(new(httpPrivateData))
+
+	private = http.Parse(&req, tcptuple, 0, private) <3>
+	private = http.ReceivedFin(tcptuple, 0, private)
+
+	private = http.Parse(&resp, tcptuple, 1, private)
+
+	logp.Debug("http", "Now sending gap..")
+
+	private, drop := http.GapInStream(tcptuple, 1, 10, private)
+	assert.Equal(t, false, drop)
+
+	private = http.ReceivedFin(tcptuple, 1, private)
+
+	trans := expectTransaction(t, http) <4>
+	assert.NotNil(t, trans)
+	assert.Equal(t, trans["notes"], []string{"Packet loss while capturing the response"})
+}
+----------------------------------------------------------------------
+
+<1> It's useful to initialize the logging system in case the `-v` flag is passed
+to `go test`. This makes it easy to get the logs for a failing test while
+keeping the output clean on a normal run.
+
+<2> Define the data we'll be using in the test.
+
+<3> Call the interface functions exported by the module. The `private` structure
+is passed from one call to the next like the TCP layer would do.
+
+<4> The
+https://github.com/elastic/packetbeat/blob/b9173ae034581205ed4853c6fb040ea5357a5c28/protos/http/http_test.go#L1182[`expectTransaction`]
+function tries to read from the `results` channel and errors the test case if
+there's no transaction present.
+
+To check the coverage of your unit tests, run the `make cover` command at the
+top of the repository.
+
+==== Integration testing
+
+Because the main input to Packetbeat are packets and the main output are JSON
+objects, a convenient way of testing its functionality is by providing PCAP
+files as input and checking the results in the files created by using the "file"
+output plugin.
+
+This is the approach taken by the tests in the
+https://github.com/elastic/packetbeat/tree/master/tests[`tests`] directory. The
+tests are written in Python and executed using
+https://nose.readthedocs.org/en/latest/[nose]. Here is a simple example test
+from the MognoDB suite:
+
+
+[source,python]
+----------------------------------------------------------------------
+    def test_mongodb_find(self):
+        """
+        Should correctly pass a simple MongoDB find query
+        """
+        self.render_config_template( <1>
+            mongodb_ports=[27017]
+        )
+        self.run_packetbeat(pcap="mongodb_find.pcap", <2>
+                            debug_selectors=["mongodb"])
+
+        objs = self.read_output() <3>
+        o = objs[0]
+        assert o["type"] == "mongodb"
+        assert o["method"] == "find"
+        assert o["status"] == "OK"
+----------------------------------------------------------------------
+
+<1> The configuration file for each test run is generated from the template. If
+your protocol plugin has options in the configuration file, you should add them
+to the template.
+
+<2> The `run_packetbeat` function receives the PCAP file to run.
```

Comment:
The run_packet routine searches for pcaps in the pcap directory.

Test are run in packetbeat/run/<test_file_name>.Test.<method>